app/connectors/s3/S3UploadFormGenerator.scala

```diff
@@ -81,7 +81,9 @@ class S3UploadFormGenerator(
 
     val contentTypeField = uploadParameters.expectedContentType.map(contentType => "Content-Type" -> contentType)
 
-    fields ++ sessionCredentials ++ metadataFields ++ contentTypeField
+    val successRedirect = uploadParameters.successRedirect.map(url =>  Map("success_action_redirect" -> url)).getOrElse(Map.empty)
```

Comment:
There's a mix of Map[String, String] and Option[(String,String)] (relying on implicit conversion) - maybe settle on one approach?